### ISSUE TITLE
Inference accuracy depends on batch size

### ISSUE THREAD

**Issue by HwMohanLiu** [tensorflow/tensorflow#19548]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: r1.8
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: 0.11.0
- **GCC/Compiler version (if compiling from source)**: 4.9
- **CUDA/cuDNN version**: 9.0
- **GPU model and memory**: 16GB
- **Exact command to reproduce**: pip3 install tensorflow_gpu-1.8.0-cp35-cp35m-manylinux1_x86_64.whl

### Describe the problem
The inference accuracy of a trained model for image recognition, e.g. ResNet-18, Incpetion_v3, depends on the batch size. For example, if the batch size of the input is set to one, the accuracy is around 0.2. However, if feed 32 images at ones, the accuracy becomes over 0.82... If feed even more images at ones, the accuracy can even increase... This is really strange.

### Source code / logs
The model will classify 16 categories.  Images for training and validating are subsets of imagenet. (Over 223,918 training images and 22, 016 validating images)

1. The input size for training is set as:
```
tf.placeholder(tf.float32, [None im_height, im_width, n_channel], name = "input_x")
```

2. data is feeded by dataset  from tfrecord. 
```
dataset = tf.data.TFRecordDataset(filename)
dataset = dataset.map(tfrecord_parser).prefetch(buffer_size = 4 * batch_size).batch(batch_size)
```

3. model loading from pb file
```
def load_pb_graph(frozen_graph_file):
    with tf.gfile.GFile(frozen_graph_file, 'rb') as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())
        
    with tf.Graph().as_default() as graph:
        tf.import_graph_def(graph_def, name="")
    
    return graph, graph_def
```

4. Computation of accuracy
```
with g2.as_default():        
    in_logits = tf.placeholder(tf.float32, [None, FLAGS.n_class])
    in_labels = tf.placeholder(tf.float32, [None, FLAGS.n_class])
    accuracy = tf.metrics.accuracy(tf.argmax(in_logits, 1), tf.argmax(in_labels, 1))[1]
        
    sess2 = tf.Session(graph = g2, config = config)
    sess2.run(tf.global_variables_initializer())
    sess2.run( tf.local_variables_initializer())
....
_logits = sess.run(logits, feed_dict = val_feed_dict) 
_accuacy = sess2.run(accuracy, {in_logits: _logits, in_labels: val_batch_y})
```

Feeding one image for inference:
> 2018-05-24 21:14:36 Step 22016 : Validation Accuracy:0.21825 

Feeding 16 images for inference：
> 2018-05-24 21:16:22 Step 22016 : Validation Accuracy:0.80069

Feeding 32 images for inference :
> 2018-05-24 21:18:14 Step 22016 : Validation Accuracy:0.81486

Feeding 128 images for inference:
> 2018-05-24 21:22:42 Step 22016 : Validation Accuracy:0.82744

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bignamehyp** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since we don't help debug custom code here. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by SunnyDjinn** [comment]
I have the same problem regarding the accuracy. When training with a greater batch size (e.g. 128), the reported accuracy is around 95% (I am using the resnet model supplied by tensorflow on a subset of the imagenet dataset). Here is the filled issue template.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04.4 LTS (Xenial Xerus)
- **TensorFlow installed from (source or binary)**: binary 
- **TensorFlow version (use command below)**: v1.8.0-0-g93bc2e2072 1.8.0
- **Python version**: Python 3.5.2
- **Bazel version (if compiling from source)**: 
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: cuda-9.0 / libcudart.so.9.0.176
- **GPU model and memory**: Tesla K80, 11441MiB
- **Exact command to reproduce**: train the resnet18 model supplied by TensorFlow here: https://github.com/tensorflow/models/tree/master/official/resnet with batch size 128 for example. After some epochs, the accuracy reported is about 95%. Then, train the same model taking the last checkpoint for 1 epoch with batch size 1. The accuracy drops down to 15%.

### Describe the problem
The reported accuracy drops down after only 1 epoch of training at batch size 1 when the model has previously been trained enough at batch size 128 to get an accuracy of 95%. This seems like an issue, either regarding the reported accuracy for validation, or regarding the training with different batch sizes. The latter seems less likely since only 1 epoch of training at batch size 1 has been made.

### Source code / logs
After many epochs of training at batch size 128:
```
06:59:31.217821 139740337723136 tf_logging.py:116] Saving dict for global step 1671932: accuracy = 0.9535984
```

Adding 1 epoch at batch size 1:
```
06:59:31.217821 139740337723136 tf_logging.py:116] Saving dict for global step 1671932: accuracy = 0.9535984
```

**Comment by bignamehyp** [comment]
Did your inference include all the validation images?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by wt-huang** [comment]
@HwMohanLiu Among other things batch size does impact accuracy to some extent, but I wouldn't use batch size 1.
